### PR TITLE
 SDK-1401 : Fix NPM Vulnerabilities Audit fix

### PR DESCRIPTION
**Pull request by JagadeeshKaricherla-branch**
# Description

- This change upgrades react-native and its dependencies
- Changes done ( manually) based on https://react-native-community.github.io/upgrade-helper/?from=0.62.0&to=0.63.0
- made changes to eslinting

## Motivation and Context
SDK-1401 (Fix NPM Vulnerabilities) needed upgrading react-native & eslint & their dependencies.I followed the guide here - https://reactnative.dev/docs/upgrading. Auto upgrade didn’t work out because of the nature of the project so I had to do them manually. Also, upgrading from 0.62 to 0.69.0 is bringing a large number of changes so instead I am doing a incremental upgrade of 1 major version at a time.

## Type of change
 :white_check_mark: feature

## How Has This Been Tested?

:white_check_mark: unit test
:white_check_mark: linted

## Checklist:

-  :white_check_mark: My code follows the style guidelines of this project
-  :white_check_mark: I have performed a self-review of my code
-  :white_check_mark: I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,26 +1,41 @@
 apply plugin: 'com.android.library'
 import groovy.json.JsonSlurper
 
+def DEFAULT_COMPILE_SDK_VERSION = 29
+def DEFAULT_MIN_SDK_VERSION = 16
```
Comment (gdeluna-branch) [inline android/build.gradle:5]: Let's bump this to `21` to align with the latest native gradle min sdk

**Comment by gdeluna-branch** [comment]
Can you run `npm audit fix`? There are some more non-breaking critical vulnerabilities that can be updated.

**Comment by JagadeeshKaricherla-branch** [comment]
> 

Upgrading to the latest version of React native (0.69.0) from 0.62.2 needs a lot of changes which are difficult to test so I am upgrading 1 major version at a time. `npm audit` still shows few vulnerabilities which will be fixed as part of the incremental updates.

File: android/build.gradle [hunk 2]
```diff
@@ -1,26 +1,41 @@
...
+    maven {
+        // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+        url "$rootDir/../node_modules/react-native/android"
+    }
+    maven {
+        // Android JSC is installed from npm
+        url "$rootDir/../node_modules/jsc-android/dist"
+    }
+    google()
+    jcenter()
```
[the remark this answers is not in the document]
Comment (JagadeeshKaricherla-branch) [reply android/build.gradle]: good call out.done.

File: android/build.gradle [hunk 3]
```diff
@@ -1,26 +1,41 @@
...
 }
 
 def safeExtGet(prop, fallback) {
     rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
-}
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
```
[the remark this answers is not in the document]
Comment (JagadeeshKaricherla-branch) [reply android/build.gradle:32]: fixed now.

File: android/build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (gdeluna-branch) [reply android/build.gradle:5]: Then it's fine for now since the native branch sdk will take precedence with higher min.

**Comment by JagadeeshKaricherla-branch** [comment]
are we good to merge this PR @gdeluna-branch @echo-branch ?

**Pull request merged**